### PR TITLE
zaaCheckboxArray: added toggleAll button.

### PR DESCRIPTION
**Pull request by chemezov**
### What are you changing/introducing

Hi there! I've added "toggle all" button to zaaCheckboxArray plugin. How it looks like:

image

1) I'm not sure I added the styles to the right files.
2) What do you think about this?
3) How can i build resources and compile main.css file?
4) How can i add i18n for tooltip?

### What is the reason for changing/introducing

Some of the lists are quite long. And there is a need to select all but a few.

### QA

| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes/no
| Fixed issues  |

### CODE REVIEW

File: src/resources/js/formdirectives.js [hunk 1]
```diff
@@ -2047,6 +2057,7 @@ zaa.directive("zaaCheckboxArray", function () {
                         '<label for="{{id}}">{{label}}</label>' +
                     '</div>' +
                     '<div class="form-side">' +
+                        '<button ng-click="toggleAll()" type="button" class="ckeckbox-list-toggle-button btn btn-info btn-icon"><i class="material-icons">done_all</i></button>' +
```
Comment (TheMaaarc) [inline src/resources/js/formdirectives.js:2060]: Rename `checkbox-list-toggle-button` to `zaa-checkbox-array-toggle-all`, we already have a `zaa-checkbox-array.scss` file which defines some `zaaCheckboxArray` spcific things.

File: src/resources/scss/admin.scss [hunk 2]
```diff
@@ -87,6 +87,7 @@
 @import "components/badge";
 @import "components/btn";
 @import "components/card";
+@import "components/checkbox-toggle-button";
```
Comment (TheMaaarc) [inline src/resources/scss/admin.scss:90]: Remove `checkbox-toggle-button` entry and file, move code to end of  `components/_zaa-checkbox-array.scss`.

File: src/resources/scss/components/_checkbox-list-toggle-button.scss [hunk 3]
```diff
@@ -0,0 +1,11 @@
+.ckeckbox-list-toggle-button {
```
Comment (TheMaaarc) [inline src/resources/scss/components/_checkbox-list-toggle-button.scss:1]: Rename to `.zaa-checkbox-array-toggle-all`.

File: src/resources/scss/components/_checkbox-list-toggle-button.scss [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  float: right;
+  margin-top: 2px;
+  position: relative;
+  right: 0;
+}
+
+.ckeckbox-list-toggle-button + luya-checkbox-array {
```
Comment (TheMaaarc) [inline src/resources/scss/components/_checkbox-list-toggle-button.scss:8]: Rename to `.zaa-checkbox-array-toggle-all + luya-checkbox-array`.

**Review by TheMaaarc — CHANGES_REQUESTED** [review]
Thanks!

**Comment by chemezov** [comment]
@TheMaaarc Thank you very much for the detailed review, much appreciated. All things done.

**Comment by chemezov** [comment]
So, there is last questions:
1) Resources: How can i build resources and compile main.css file?
2) Tooltip: How can i add i18n for tooltip?
3) What to do with tests? I have very little experience with this.

**Comment by nadar** [comment]
> Resources: How can i build resources and compile main.css file?
We will do that for you before release

> Tooltip: How can i add i18n for tooltip?
Is this related to that PR? There are tooltip directives which works with i18n text. Please create a forum entry so we can explain

> What to do with tests? I have very little experience with this.
Not that much of js tests sadly.

**Comment by chemezov** [comment]
Done fixes and added changelog entry. If tooltip is not necessary, this task is done completely.

**Comment by nadar** [comment]
> If tooltip is not necessary

I think thats ok like that :+1:

**Review by nadar — APPROVED** [review]

**Pull request merged**